kernel/buddy_allocator.cpp

```diff
@@ -0,0 +1,137 @@
+#include "buddy_allocator.hpp"
+
+
+#include <bolgenos-ng/memory.hpp>
+
+
+memory::allocators::BuddyAllocator::~BuddyAllocator() {
+}
+
+
+void memory::allocators::BuddyAllocator::initialize(
+		memory::MemoryRegion *region, size_t max_order) {
+	region_ = region;
+	max_order_ = max_order;
+	free_list_ = nullptr;
+}
+
+
+void memory::allocators::BuddyAllocator::put(pblk_t blk) {
+	if (free_list_ == nullptr) {
+		size_t bytes_for_fls = sizeof(FreeList) * max_order_;
```

Comment:
:bangbang: 
Add interface for calculating the size of buddy allocator. Also buddy system order can be hard-coded or configured at compile-time. Anyway, it's bad idea to initialize on the first put.